serverloop.c

```diff
@@ -194,15 +193,11 @@ wait_until_can_do_something(struct ssh *ssh,
 	 * analysis more difficult, but we're not doing it yet.
 	 */
 	if (options.client_alive_interval) {
-		uint64_t keepalive_ms =
-		    (uint64_t)options.client_alive_interval * 1000;
-
-		if (max_time_ms == 0 || max_time_ms > keepalive_ms) {
-			max_time_ms = keepalive_ms;
-			client_alive_scheduled = 1;
-		}
 		if (last_client_time == 0)
 			last_client_time = monotime();
+		ptimeout_deadline_monotime(&timeout,
+		    last_client_time + options.client_alive_interval);
```

Comment:
would ptimeout_deadline_sec(&timeout,options.client_alive_interval);
match the old behaviour better?

Reply:
Yes, though I think this way is more accurate. In the spirit of changing only one thing at a time, I've restored the original behaviour and left an XXX comment to remind me to come back to it :)